src/ts/common.ts

```diff
@@ -1156,6 +1156,9 @@ on('autocomplete', (el: HTMLElement) => {
   if (el.dataset.target === 'users') {
     transformer = user => `${user.userid} - ${user.fullname} (${user.email})`;
   }
+  if (el.dataset.target === 'compounds') {
+    transformer = compound => `${compound.id} - ${compound.name}`;
```

Comment:
Add CAS in there if available. Such as id - name (CAS: XX-YY-ZZZ), otherwise id - name.